src/isar/state_machine/states/monitor.py

```diff
@@ -70,6 +75,35 @@ def _run(self) -> None:
                 time.sleep(self.state_machine.sleep_time)
                 continue
 
+            except RobotCommunicationTimeoutException as e:
+                self.state_machine.current_mission.error_message = ErrorMessage(
+                    error_reason=e.error_reason, error_description=e.error_description
+                )
+                self.logger.warning(
+                    f"Monitoring step {self.state_machine.current_step.id} failed #: "
+                    f"{self.request_status_failure_counter} failed because: {e.error_description}"
+                )
+
+                self.step_status_thread = None
+                self.request_status_failure_counter += 1
```

Comment:
Is this ever reset to 0?

Reply:
I guess we want it to be reset as soon as we have successfully retried

Reply:
No, I have added this now.